testsuite/compare.cmake

```diff
@@ -0,0 +1,27 @@
+# This is used to compare to files using CMake.
+# It improves on the default behaviour to print the
+# the files if there is a difference.
+
+execute_process(
+    COMMAND ${CMAKE_COMMAND} -E compare_files --ignore-eol ${original_file} ${new_file}
+    RESULT_VARIABLE status
+)
+
+if (${status} EQUAL 1)
+    message ("Compare ${original_file} with ${new_file}")
+    file(READ ${original_file} original_text)
+    file(READ ${new_file} new_text)
+    message("--Original File-----------------------------------------------------------------")
+    if (NOT original_text STREQUAL "")
+        message("${original_text}")
+    endif()
+    message("--------------------------------------------------------------------------------")
+    message("  ")
+    message("--New File----------------------------------------------------------------------")
+    if (NOT new_text STREQUAL "")
+        message(${new_text})
+    endif()
+    message("--------------------------------------------------------------------------------")
+
+    message(FATAL_ERROR "Files differ!")

```

Comment:
There is no guarantee that the files are different when the process returns an error. What about tests that do check for errors and that's actually a pass?

Reply:
Could you elaborate please?

Reply:
As I said above, I mistook this for the program execution. this is just the result of the compare, which is 1, is different.